### PR TITLE
Run tests on PKS

### PR DESCRIPTION
**Pull request by sbawaska**
Spins up a new pks cluster for running tests.

### CODE REVIEW

File: util.sh [hunk 1]
```diff
@@ -123,7 +123,12 @@ fats_setup_gcloud() {
   sudo apt-get update && sudo apt-get install google-cloud-sdk
 
   gcloud config set project cf-spring-pfs-eng
+  gcloud config set compute/region us-central1
   gcloud config set compute/zone us-central1-a
+  # TODO debug why config set for region and zone does not work
+  # current workaround is to use the env variables below
+  export CLOUDSDK_COMPUTE_REGION="us-central1"
```
Comment (scothis) [inline util.sh]: we should be able to lookup these value with gcloud when we need them, `gcloud config get-value compute-region`

File: install_pks.sh [hunk 2]
```diff
@@ -22,10 +22,16 @@ export UAA_ADMIN_PASSWORD=$(echo $TOOLSMITH_ENV | base64 --decode | jq -r .pks_a
 
 pks login -a api.pks.${TS_G_ENV}.cf-app.com -u admin -p ${UAA_ADMIN_PASSWORD} -k
 
-# TODO uncomment if we use a PKS cluster per job
-# pks create-cluster ${TS_G_ENV}-${CLUSTER_NAME} --external-hostname ${CLUSTER_NAME}.${TS_G_ENV}.cf-app.com --plan small --wait
+gcloud compute addresses create ${TS_G_ENV}-${CLUSTER_NAME}-ip --region=${CLOUDSDK_COMPUTE_REGION}
```
Comment (scothis) [inline install_pks.sh]: we shouldn't need to specify the region if it is set in the gcloud config. Unless the region is required for this command and the config is ignored.

File: install_pks.sh [hunk 3]
```diff
@@ -22,10 +22,16 @@ export UAA_ADMIN_PASSWORD=$(echo $TOOLSMITH_ENV | base64 --decode | jq -r .pks_a
...
+export LB_IP=`gcloud compute addresses list --filter="name=(${TS_G_ENV}-${CLUSTER_NAME}-ip)" --format=json | jq -r .[0].address`
```
Comment (scothis) [inline install_pks.sh]: our convention for local var is to not export them and lowercase

File: install_pks.sh [hunk 4]
```diff
@@ -22,10 +22,16 @@ export UAA_ADMIN_PASSWORD=$(echo $TOOLSMITH_ENV | base64 --decode | jq -r .pks_a
...
+
+pks create-cluster ${TS_G_ENV}-${CLUSTER_NAME} --external-hostname ${LB_IP} --plan large --wait
```
Comment (scothis) [inline install_pks.sh]: I assume you're skipping setting up dns, by using the ip directly here instead of a hostname, nice.

File: cleanup_pks.sh [hunk 5]
```diff
@@ -2,5 +2,12 @@
 
 source `dirname "${BASH_SOURCE[0]}"`/util.sh
 
-# TODO uncomment if we use a PKS cluster per job
-# pks delete-cluster ${TS_G_ENV}-${CLUSTER_NAME} --non-interactive --wait
+fats_setup_gcloud
```
Comment (scothis) [inline cleanup_pks.sh]: gcloud is already setup, unless you're trying to pickup the envvar which I'm not convinced we need (see below)

File: cleanup_pks.sh [hunk 6]
```diff
@@ -2,5 +2,12 @@
...
+
+export TS_G_ENV=$(echo $TOOLSMITH_ENV | base64 --decode | jq -r .name)
+gcloud compute forwarding-rules delete ${TS_G_ENV}-${CLUSTER_NAME}-fw --region=${CLOUDSDK_COMPUTE_REGION}
+gcloud compute target-instances delete ${TS_G_ENV}-${CLUSTER_NAME}-ti --zone=${CLOUDSDK_COMPUTE_ZONE}
+gcloud compute addresses delete ${TS_G_ENV}-${CLUSTER_NAME}-ip --region=${CLOUDSDK_COMPUTE_REGION}
+
+pks delete-cluster ${TS_G_ENV}-${CLUSTER_NAME} --non-interactive --wait
```
Comment (scothis) [inline cleanup_pks.sh]: we may need `travis_wait` here, but let's see if it fails before we add it

File: .travis.yml [hunk 7]
```diff
@@ -5,10 +5,11 @@ go:
 env:
   global:
   - CLUSTER_NAME="fats-${TRAVIS_JOB_NUMBER//./-}"
+  - NAMESPACE="fats-${TRAVIS_JOB_NUMBER//./-}"
 install: true
 before_script:
 - "./install.sh kubectl"
-- "./install.sh $CLUSTER"
+- "travis_wait 60 ./install.sh $CLUSTER"
```
[the remark this answers is not in the document]
Comment (sbawaska) [reply .travis.yml]: travis is going to kill the job after 50 minutes no matter how long we set this time to. (50 mins for public and 120 mins for private)

**Comment by sbawaska** [comment]
closing in favor of #66.

**Pull request closed without merging**